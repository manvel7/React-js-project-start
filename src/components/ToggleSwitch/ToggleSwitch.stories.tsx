import React, {useState} from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import ToggleSwitch from "./ToggleSwitch"

const meta: Meta = {
  title: "Components/ToggleSwitch",
  component: ToggleSwitch
}

const ToggleSwitchStory: Story = (args) => {
  const [checked, setChecked] = useState(false)

  const onChange = () => {
    setChecked((prev) => !prev)
  }

  return (
    <ToggleSwitch
      {...args}
      checked={checked}
      onChange={onChange}
      label="Switch"
    />
  )
}

const Default = ToggleSwitchStory.bind({})

const Disabled = ToggleSwitchStory.bind({})
Disabled.args = {
  disabled: true
}
const ErrorState = ToggleSwitchStory.bind({})
ErrorState.args = {
  error: true
}

export {Default, Disabled, ErrorState}

export default meta
