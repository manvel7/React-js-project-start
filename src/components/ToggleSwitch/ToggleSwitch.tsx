import React, {FunctionComponent, HTMLAttributes, ReactNode} from "react"
import ToggleSwitchContainer from "./Container"
import HiddenCheckbox from "./HiddenCheckbox"
import VisibleSwitch from "./VisibleSwitch"

export interface IProps extends HTMLAttributes<HTMLInputElement> {
  checked?: boolean
  disabled?: boolean
  error?: boolean
  label?: ReactNode
}

const ToggleSwitch: FunctionComponent<IProps> = (props) => {
  const {checked, disabled, label, error} = props

  return (
    <ToggleSwitchContainer>
      <HiddenCheckbox {...props} />
      <VisibleSwitch
        checked={checked}
        disabled={disabled}
        error={error}
        hasLabel={!!label}>
        <div className="toggle-switch__dot" />
      </VisibleSwitch>
      {label}
    </ToggleSwitchContainer>
  )
}

export default ToggleSwitch
