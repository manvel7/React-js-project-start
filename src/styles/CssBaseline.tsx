import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent,
  StyledProps
} from "styled-components"
import {mediaBreakpointDown} from "../utils/breakpoint"
import {getValue, toPercent} from "../utils/unit"
import {
  bodyMixin,
  smallMixin,
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin
} from "./typography"

interface IProps {}

const genericResetMixin = css<StyledProps<IProps>>`
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`

const appResetMixin = css<StyledProps<IProps>>`
  html {
    /* stylelint-disable property-no-vendor-prefix */
    -ms-overflow-style: scrollbar; /* Set box-sizing so that the width is not affected by padding or border */
    -ms-text-size-adjust: 100%; /* Set default font family in all browsers. */
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); /* Makes 1rem = 10px */
    -webkit-text-size-adjust: 100%; /* Prevent adjustments of font size after orientation changes in IE on Windows Phone and in iOS. */
    /* stylelint-enable property-no-vendor-prefix */
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    background-color: ${({theme}) => theme.palette.background.white};

    /* Hide overflow content */
    &.overflow {
      overflow: hidden;
    }

    /* Prevent focus outline on elements that can't be accessed via keyboard. */
    [tabindex="-1"]:focus {
      outline: none !important;
    }

    /* Hide overflow in IE */
    svg:not(:root) {
      overflow: hidden;
    }

    /* Hide elements has 'hidden' attribute in IE 10- */
    [hidden] {
      display: none !important;
    }

    ::-webkit-file-upload-button {
      /* stylelint-disable-next-line property-no-vendor-prefix */
      -webkit-appearance: button;
      font: inherit;
    }
  }
`

const typographyMixin = css<StyledProps<IProps>>`
  html {
    font-family: ${({theme}) =>
      theme.typography
        .fontFamily}; /* Prevent scrollbars to overlap content in IE11 and Edge */
    font-size: ${({theme}) =>
      toPercent(
        getValue(theme.typography.htmlFontSize),
        16
      )}; /* Set default tap highlight to be transparent in iOS. */
    font-style: normal;
    color: ${({theme}) => theme.palette.text.dark};
  }

  body {
    ${bodyMixin<IProps>()}
  }

  h1 {
    ${h1Mixin<IProps>()}
  }

  h2 {
    ${h2Mixin<IProps>()}
  }

  h3 {
    ${h3Mixin<IProps>()}
  }

  h4 {
    ${h4Mixin<IProps>()}
  }

  h5 {
    ${h5Mixin<IProps>()}
  }

  h6 {
    ${h6Mixin<IProps>()}
  }

  small {
    ${smallMixin<IProps>()}
  }

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.bold};

    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    caption {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

const utilsMixin = css<StyledProps<IProps>>`
  .hidden {
    visibility: hidden;
  }

  .d-none {
    display: none;
  }

  .hidden-sm-down {
    ${({theme}) =>
      mediaBreakpointDown(
        "sm",
        theme.grid.breakpoints,
        `
      display: none;
    `
      )}
  }

  .text-center {
    text-align: center;
  }
`

const CssBaseline: GlobalStyleComponent<
  IProps,
  DefaultTheme
> = createGlobalStyle`
  ${genericResetMixin}

  ${appResetMixin}

  ${typographyMixin}

  ${utilsMixin}
`

export default CssBaseline
