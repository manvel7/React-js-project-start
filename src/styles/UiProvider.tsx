import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
