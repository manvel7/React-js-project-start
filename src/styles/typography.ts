import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  StyledProps,
  ThemedStyledProps
} from "styled-components"
import {pxToRem} from "../utils/unit"

const h1Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h1.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h1.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h1.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h1.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h2Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h2.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h2.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h2.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h2.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h3Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h3.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h3.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h3.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h3.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h4Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h4.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h4.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h4.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h4.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h5Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h5.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h5.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h5.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h5.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h6Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h6.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h6.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h6.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h6.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const bodyMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.body.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.body.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.lineHeight,
      theme.typography.htmlFontSize
    )};
`

const smallMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.caption.weight};
  letter-spacing: ${({theme}) =>
    theme.typography.variants.caption.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`
export {
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin,
  bodyMixin,
  smallMixin
}
