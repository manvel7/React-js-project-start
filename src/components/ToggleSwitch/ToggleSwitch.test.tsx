import React from "react"
import {render} from "@testing-library/react"
import {UiProvider} from "../../styles"
import ToggleSwitch from "./ToggleSwitch"

describe("ToggleSwitch", () => {
  describe("Default", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <ToggleSwitch checked={false} />
        </UiProvider>
      )

      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("Default checked", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <ToggleSwitch checked={true} />
        </UiProvider>
      )

      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("Disabled", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <ToggleSwitch checked={false} disabled={true} />
        </UiProvider>
      )

      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("Error", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <ToggleSwitch checked={false} error={true} />
        </UiProvider>
      )

      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
