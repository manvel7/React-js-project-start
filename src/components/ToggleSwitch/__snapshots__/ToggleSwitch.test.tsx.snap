// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Default checked should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.8rem;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  background-color: #E6E6E6;
  border-radius: 0.4rem;
  display: inline-block;
  height: 0.8rem;
  margin-right: 0;
  position: relative;
  -webkit-transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  width: 3.2rem;
  background-color: #001254;
}

.c2 .toggle-switch__dot {
  background-color: #001254;
  border-radius: 50%;
  height: 1.6rem;
  left: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -ms-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -webkit-transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  width: 1.6rem;
}

.c2 .toggle-switch__dot {
  left: calc(100% - 0.8rem);
}

<label
  class="c0"
>
  <input
    checked=""
    class="c1"
    type="checkbox"
  />
  <div
    class="c2"
  >
    <div
      class="toggle-switch__dot"
    />
  </div>
</label>
`;

exports[`ToggleSwitch Default should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.8rem;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  background-color: #E6E6E6;
  border-radius: 0.4rem;
  display: inline-block;
  height: 0.8rem;
  margin-right: 0;
  position: relative;
  -webkit-transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  width: 3.2rem;
}

.c2 .toggle-switch__dot {
  background-color: #001254;
  border-radius: 50%;
  height: 1.6rem;
  left: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -ms-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -webkit-transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  width: 1.6rem;
}

<label
  class="c0"
>
  <input
    class="c1"
    type="checkbox"
  />
  <div
    class="c2"
  >
    <div
      class="toggle-switch__dot"
    />
  </div>
</label>
`;

exports[`ToggleSwitch Disabled should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.8rem;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  background-color: #E6E6E6;
  border-radius: 0.4rem;
  display: inline-block;
  height: 0.8rem;
  margin-right: 0;
  position: relative;
  -webkit-transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  width: 3.2rem;
  background-color: #DDDDDD;
}

.c2 .toggle-switch__dot {
  background-color: #001254;
  border-radius: 50%;
  height: 1.6rem;
  left: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -ms-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -webkit-transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  width: 1.6rem;
}

.c2 .toggle-switch__dot {
  background-color: #808080;
}

<label
  class="c0"
>
  <input
    class="c1"
    disabled=""
    type="checkbox"
  />
  <div
    class="c2"
    disabled=""
  >
    <div
      class="toggle-switch__dot"
    />
  </div>
</label>
`;

exports[`ToggleSwitch Error should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.8rem;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  background-color: #E6E6E6;
  border-radius: 0.4rem;
  display: inline-block;
  height: 0.8rem;
  margin-right: 0;
  position: relative;
  -webkit-transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: background-color 0.1s cubic-bezier(0.47,0.09,0.86,0.68);
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  width: 3.2rem;
  background-color: #E6E6E6;
}

.c2 .toggle-switch__dot {
  background-color: #001254;
  border-radius: 50%;
  height: 1.6rem;
  left: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -ms-transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  transform: translateY(-50%) translateX(calc(-50% + 0.4rem));
  -webkit-transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  transition: left 0.2s cubic-bezier(0.47,0.09,0.86,0.68);
  width: 1.6rem;
}

.c2 .toggle-switch__dot {
  background-color: #CA0000;
}

<label
  class="c0"
>
  <input
    class="c1"
    type="checkbox"
  />
  <div
    class="c2"
  >
    <div
      class="toggle-switch__dot"
    />
  </div>
</label>
`;
